internal/source/artifact_image.go

```diff
@@ -0,0 +1,49 @@
+package source
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+
+	oras "oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+const unpackLocation = "/tmp/"
+
+type ArtifactImage struct {
+}
+
+func (i *ArtifactImage) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeOCIArtifacts {
+		return nil, fmt.Errorf("bundle source type %q not supported", bundle.Spec.Source.Type)
+	}
+	if bundle.Spec.Source.Image == nil {
+		return nil, fmt.Errorf("bundle source image configuration is unset")
+	}
+	tmpfs, err := file.New(unpackLocation)
+	if err != nil {
+		return nil, err
+	}
+	defer tmpfs.Close()
+	refComponents := strings.Split(bundle.Spec.Source.Image.Ref, ":")
+	repo, err := remote.NewRepository(refComponents[0])
+	if err != nil {
+		return nil, err
+	}
+	// 2. Copy from the remote repository to the file store
+	manifestDescriptor, err := oras.Copy(ctx, repo, refComponents[1], tmpfs, refComponents[1], oras.DefaultCopyOptions)
```

Comment:
I'm curious if there is a chance that this call blocks the reconciliation for an extended period of time. I know that the OCI artifacts are meant to be on the smaller size for pulling anyways, but I can imagine a scenario where someone's OCI artifact bundle is fairly large and pulling it takes significantly longer than usual or expected.

IMO we should probably take this into consideration and prevent this from happening. Off the top of my head, one way to do this would be to use unpack `Job`s similar to how the current `image` source unpacks bundles. The only difference here is that in the `Pod` created by the `Job` we could use something like the `oras` cli to copy the bundle information to a local directory and then run the same `/bin/unpack` command the `image` source does. While this is a significantly more complex implementation than you have here, I think the biggest benefit is that we eliminate a case where reconciliation gets blocked and holds up the reconciliation process for other `BundleDeployment`s.

One thing that I think could be useful to prevent a large amount of code duplication is to update the `image` source to maybe instead be an `oci` source that can work with both images *and* OCI artifacts and make adjustments accordingly.